### ISSUE TITLE
ThemeProvider probably shouldn't return "div"

### ISSUE THREAD

**Issue by danielmahon** [jamesmfriedman/rmwc#317]
The `ThemeProvider` probably shouldn't return a new `div` as this can mess with styling things like 100% height, flex, grid content, and would need to be manually adjusted. Can we convert to `Fragment` or more backward compatible solution? Maybe similar to how it's handled in `styled-components`?
https://github.com/styled-components/styled-components/blob/8102cca24a97a89ed1444283d1e5e4c410c9e5c2/src/models/ThemeProvider.js

I haven't messed with a solution yet, as I pretty much just do my theming with Sass so I can have better control over MDC, but If I get some time I'll take a stab at it.

https://github.com/jamesmfriedman/rmwc/blob/d9008f9053a236f371ec2800b927ecb7b8f057b7/src/theme/index.js#L80

**Comment by jamesmfriedman** [comment]
It actually has to be a div @danielmahon, but we might be able to do something to make it better.

- We can't use a fragment because we need to have the styles injected into the DOM
- We can't use a <style> tag because no matter where you put them, they are considered global
- We HAVE to use some DomNode because of the way CSS variables cascade.

A subtle and hidden feature in a lot of RMWC is the ability to "wrap" a component. This basically takes the props from a component, and applies them to the components child. An example.

```jsx
<ThemeProvider options={{...}} wrap>
  <TabBar />
</ThemeProvider>
```

Would collapse the output of ThemeProvider onto its child element. In this case, adding the style attribute.

```jsx
<div class="mdc-tab-bar" style={...}></div>
```

This seems like it would be a good improvement and solve  your problem, yes?

**Comment by danielmahon** [comment]
@jamesmfriedman That should™ work! I did try using the wrap attribute but noticed after checking the code it wasn't supported on `ThemeProvider`. My initial use case had it a sibling of my `styled-components` ThemeProvider so not sure if wrap would work if parented to other non-dom components like `BrowserRouter`, etc, but it could probably be moved around a bit if need be. 

      <ContextProvider value={this.state}>
        <StyledThemeProvider theme={theme}>
          <RMWCThemeProvider options={theme.rwmc}>
            <BrowserRouter>
            ...

**Comment by jamesmfriedman** [comment]
Will be released with 3.0.0
